src/io/flutter/module/settings/ProjectType.java

```diff
@@ -95,11 +110,52 @@ public ComboBox getProjectTypeCombo() {
     return projectTypeCombo;
   }
 
-  public void setSdk(@NotNull FlutterSdk sdk) {
-    this.sdk = sdk;
+  public void setSdk(@NotNull Supplier<? extends FlutterSdk> sdk) {
+    this.getSdk = sdk;
+    adjustAndroidX();
   }
 
   public void addListener(ItemListener listener) {
     projectTypeCombo.addItemListener(listener);
   }
+
+  public void cacheState() {
+    if (getType() != FlutterProjectType.PACKAGE) {
+      androidxCheckboxValue = androidxCheckbox.isSelected();
+    }
+  }
+
+  public void adjustAndroidX(@Nullable Project project) {
+    if (project != null) {
```

Comment:
nit: seems surprising to have the override with one argument call the version with zero arguments.

I do worry that this logic as written may have bugs as adjustAndroidX can never promote the checkbox to visible from invisible but perhaps that is by design.

Reply:
Right, visibility is determined by the Flutter SDK. ProjectType can control selection and enablement but visibility is handled in FlutterCreateAdditionalSettingsFields, which has access to the SDK. The zero-arg method should have been private; changed.